[Pages] update framework guides to reference C3 (#10357)
cloudflare/cloudflare-docs: content/pages/framework-guides/deploy-an-astro-site.md

## Setting up a new project

> The quickest way to get started is to use the create-cloudflare CLI (c3) tool. This will create a new project directory, initiate Astro's official setup tool, and provide the option to deploy instantly.
> 
> To use create-cloudflare to create a new Astro project, run the following command:

Reviewer: ```suggestion
Use the [`create-cloudflare`](https://www.npmjs.com/package/create-cloudflare) CLI (C3) to set up a new project. C3 will create a new project directory, initiate Astro's official setup tool, and provide the option to deploy instantly.

To use `create-cloudflare` to create a new Astro project, run the following command:
```

```sh
$ npm create cloudflare@latest my-astro-app -- --framework=astro
```

Astro will ask:

1. Which project type you would like to set up. Your answers will not affect the rest of this tutorial. Select an answer ideal for your project.

> 2. If you want to set initialize a Git repository. We recommend you to select `No` and follow this guide's [Git instructions](/pages/framework-guides/deploy-an-astro-site/#create-a-github-repository) below. If you select `Yes`, do not follow the below Git instructions precisely but adjust them to your needs.

Author: Yeah..the `set` shouldn't be there.
